### DISCUSSION TITLE
dynamic change of src

### DISCUSSION THREAD

**Discussion by AgustinMaurel** [google/model-viewer#3674] [Q&A]
hello i am having the following problem when i want to change the src of the model viewer, by means of a button it changes a local state which is heard by a useeffect and is updated in the viewer but with the first one it renders well. when i change it it does not show me the new model and i get the following error, the models are saved in an ipfs, i am doing it with Javascript / React. Thanks ! 
![CapturaErr](https://user-images.githubusercontent.com/85203984/182956310-34e60d90-39a0-4f81-8b7c-a29d867e01cd.PNG)

**Comment by elalish** [comment]
You need to set up your server to supply CORS headers that allow your origin to access the file from JS. This isn't necessary for images (since the browser handles those), but is necessary for GLBs.

**Reply by AgustinMaurel** [reply]
i am saving the models in ipfs ( pinata ) and my server sends me the url, sorry but i don't understand what cors i have to set if the src is the url of the ipfs and with the first model it works, any example of the above? Thanks !

**Comment by camradex** [comment]
did you manage to figure this out?
